### PR TITLE
Add tests for message endpoint

### PR DESCRIPTION
**Pull request by brngylni**
This PR adds the tests for message endpoint

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -43,8 +43,8 @@ def create_service(client):
     with client.application.app_context():
         # Setup code to create the object in the database
         user, created = get_or_create(db.session, User, username="mehmet") # Adjust fields as necessary
-        get_or_create(db.session, Service, name="Github Demo", type="Github", desc="description", user_id=user.id)
-
+        service, created = get_or_create(db.session, Service, name="Github Demo", type="Github", desc="description", user_id=user.id)
+        service.token = "43b901cba25c458a9175ef119a9603bb"
```
Comment (gridhead) [inline tests/conftest.py:47]: Is it your actual GitHub token or the one generated by our service?

File: tests/conftest.py [hunk 2]
```diff
@@ -43,8 +43,8 @@ def create_service(client):
     with client.application.app_context():
         # Setup code to create the object in the database
         user, created = get_or_create(db.session, User, username="mehmet") # Adjust fields as necessary
-        get_or_create(db.session, Service, name="Github Demo", type="Github", desc="description", user_id=user.id)
-
+        service, created = get_or_create(db.session, Service, name="Github Demo", type="Github", desc="description", user_id=user.id)
```
Comment (gridhead) [inline tests/conftest.py:46]: GitHub is a proper noun.

```suggestion
        service, created = get_or_create(db.session, Service, name="GitHub Demo", type="GitHub", desc="description", user_id=user.id)
```

File: tests/test_message/event_sample.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
+
```
Comment (gridhead) [inline tests/test_message/event_sample.py:1]: Please remove the empty line from the top of the file.

File: tests/test_message/event_sample.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    "events_url": "https://api.github.com/users/brngylni/events{/privacy}",
+    "received_events_url": "https://api.github.com/users/brngylni/received_events",
+    "type": "User",
+    "site_admin": False
+  },
+  "created": False,
+  "deleted": False,
+  "forced": False,
+  "base_ref": None,
+  "compare": "https://github.com/brngylni/demo_repo/compare/034c31e8e464...443f68ce112a",
```
Comment (gridhead) [inline tests/test_message/event_sample.py:138]: Can this be emulated using VCR.py? 

Ref. https://vcrpy.readthedocs.io/

File: tests/test_message/event_sample.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+  "commits": [
+    {
+      "id": "443f68ce112ab055bbf48666762a626baa4729f6",
+      "tree_id": "6c91650b5abfaa946051d8c51b6e18eafddbf159",
+      "distinct": True,
+      "message": "demo",
+      "timestamp": "2024-07-23T00:16:24+03:00",
+      "url": "https://github.com/brngylni/demo_repo/commit/443f68ce112ab055bbf48666762a626baa4729f6",
```
Comment (gridhead) [inline tests/test_message/event_sample.py:146]: Can this be emulated using VCR.py?

Ref. https://vcrpy.readthedocs.io/

File: tests/test_message/event_sample.py [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+      "starred_url": "https://api.github.com/users/brngylni/starred{/owner}{/repo}",
+      "subscriptions_url": "https://api.github.com/users/brngylni/subscriptions",
+      "organizations_url": "https://api.github.com/users/brngylni/orgs",
+      "repos_url": "https://api.github.com/users/brngylni/repos",
+      "events_url": "https://api.github.com/users/brngylni/events{/privacy}",
+      "received_events_url": "https://api.github.com/users/brngylni/received_events",
+      "type": "User",
+      "site_admin": False
+    },
+    "html_url": "https://github.com/brngylni/demo_repo",
```
Comment (gridhead) [inline tests/test_message/event_sample.py:34]: Is there a way to emulate this without using a secondary external repository?

I want our test cases to be self-sufficient and not dependent on external factors.

File: tests/test_message/event_sample.py [hunk 7]
```diff
@@ -0,0 +1,204 @@
+
+request = {
+  "ref": "refs/heads/main",
+  "before": "034c31e8e46431711bed9eada103aa552aaee6dd",
+  "after": "443f68ce112ab055bbf48666762a626baa4729f6",
+  "repository": {
+    "id": 807808293,
+    "node_id": "R_kgDOMCYtJQ",
+    "name": "demo_repo",
+    "full_name": "brngylni/demo_repo",
```
Comment (gridhead) [inline tests/test_message/event_sample.py:10]: Is there a way to emulate this without using a secondary external repository?

I want our test cases to be self-sufficient and not dependent on external factors.

File: tests/test_message/event_sample.py [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+    "merges_url": "https://api.github.com/repos/brngylni/demo_repo/merges",
+    "archive_url": "https://api.github.com/repos/brngylni/demo_repo/{archive_format}{/ref}",
+    "downloads_url": "https://api.github.com/repos/brngylni/demo_repo/downloads",
+    "issues_url": "https://api.github.com/repos/brngylni/demo_repo/issues{/number}",
+    "pulls_url": "https://api.github.com/repos/brngylni/demo_repo/pulls{/number}",
+    "milestones_url": "https://api.github.com/repos/brngylni/demo_repo/milestones{/number}",
+    "notifications_url": "https://api.github.com/repos/brngylni/demo_repo/notifications{?since,all,participating}",
+    "labels_url": "https://api.github.com/repos/brngylni/demo_repo/labels{/name}",
+    "releases_url": "https://api.github.com/repos/brngylni/demo_repo/releases{/id}",
+    "deployments_url": "https://api.github.com/repos/brngylni/demo_repo/deployments",
```
Comment (gridhead) [inline tests/test_message/event_sample.py:73]: Is there a way to emulate this without using a secondary external repository?

I want our test cases to be self-sufficient and not dependent on external factors.

File: tests/test_message/event_sample.py [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+    "created_at": 1717013663,
+    "updated_at": "2024-07-22T21:14:26Z",
+    "pushed_at": 1721682985,
+    "git_url": "git://github.com/brngylni/demo_repo.git",
+    "ssh_url": "git@github.com:brngylni/demo_repo.git",
+    "clone_url": "https://github.com/brngylni/demo_repo.git",
+    "svn_url": "https://github.com/brngylni/demo_repo",
```
Comment (gridhead) [inline tests/test_message/event_sample.py:80]: Is there a way to emulate this without using a secondary external repository?

I want our test cases to be self-sufficient and not dependent on external factors.

File: tests/test_message/test_message_create.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+            404,
+            id="MESSAGE Endpoint - Not Found",
+        )
+    ]
+)
+
+@pytest.mark.usefixtures("create_service")
+def test_not_found(client, data, headers, code):
+    response = client.post("/message/2", json=data, headers=headers)
+    assert response.status_code == code
```
Comment (gridhead) [inline tests/test_message/test_message_create.py:50]: I suspect that these can be combined to be four parameters under one testing function. Simply parameterize the URL path too (as in `f"/message/{id}"`, where the `id` is provided from the parameters) and you should be fine.

**Review by gridhead — CHANGES_REQUESTED** [review]
I wonder if it is sending any messages to the staging Fedora Messaging deployment or if it is not doing anything at all in the backend. Anyway, please make the suggested changes and then we can have another round of reviews.

File: tests/conftest.py (hunk 1 above)
Comment (brngylni) [reply tests/conftest.py:47]: It was generated by our service and then I got it from there.

File: tests/test_message/event_sample.py (hunk 6 above)
Comment (brngylni) [reply tests/test_message/event_sample.py:34]: Do you mean a seperate repository under fedora infra?

File: tests/test_message/event_sample.py (hunk 4 above)
Comment (brngylni) [reply tests/test_message/event_sample.py:137]: I checked the documentation but it seems like there's a request, response structure when recording requests whereas we receive the request from GitHub to a webhook. So I'm not sure if we can make use of this. What do you think?

` import vcr
import urllib.request

with vcr.use_cassette('fixtures/vcr_cassettes/synopsis.yaml'):
    response = urllib.request.urlopen('http://www.iana.org/domains/reserved').read()
    assert b'Example domains' in response `

https://vcrpy.readthedocs.io/en/latest/usage.html#record-modes

File: tests/test_message/event_sample.py (hunk 6 above)
Comment (gridhead) [reply tests/test_message/event_sample.py:33]: @brngylni I mean by not having an external repository at all.

Comment (brngylni) [reply tests/test_message/event_sample.py:33]: I thought having a real request might be helpful in terms of testing the hash and etc. That's why I used a real payload. I can edit the payload if you'd like to.

**Pull request merged**